File: rohitMigrationTesting/pipelines/pipeline1/code/src/main/scala/io/prophecy/pipelines/pipeline1/udfs/UDFs.scala
package io.prophecy.pipelines.pipeline1.udfs

import _root_.io.prophecy.abinitio.ScalaFunctions._
import _root_.io.prophecy.libs._
import org.apache.spark.sql.types._
import org.apache.spark.sql.functions._
import org.apache.spark.sql._

object UDFs extends Serializable {
  var x = 10

  def registerUDFs(spark: SparkSession) = {
    spark.udf.register("trimString", trimString)
    spark.udf.register("addNumbers", addNumbers)
  }

  def trimString = udf((value: String) => value.trim())
  def addNumbers = udf((value: Int, value2: Int) => value + value2)
}

File: MigrationPy/pipelines/p1/code/p1/udfs/UDFs.py
from pyspark.sql import *
from pyspark.sql.functions import *
from pyspark.sql.types import *
from prophecy.lookups import (
    createLookup,
    createRangeLookup,
    lookup,
    lookup_last,
    lookup_match,
    lookup_count,
    lookup_row,
    lookup_row_reverse,
    lookup_nth
)
delimitter1 = ''

def registerUDFs(spark: SparkSession):
    spark.udf.register("udfConcat", udfConcat)

def udfConcatGenerator():
    delimitter1 = ''

    @udf(returnType = StringType())
    def func(value: str, value2: str):
        return value + delimitter1 + value2

    return func

udfConcat = udfConcatGenerator()
